tmt/cli.py

```diff
@@ -2081,3 +2087,26 @@ def completion_fish(context: Context, install: bool, **kwargs: Any) -> None:
     Setup shell completions for fish.
     """
     setup_completion('fish', install)
+
+
+@main.command(name='link')
+@pass_context
+@click.argument('link', nargs=1, metavar='[RELATION:]TARGET')
+@click.argument('names', nargs=-1, metavar='[NAME]...')
+@option(
+    '--separate', is_flag=True,
+    help="Create linking separately for multiple passed objects.")
+def link(context: Context,
+         names: list[str],
+         link: list[str],
+         separate: bool,
+         ) -> None:
+    nodes: list[Union['tmt.base.Test', 'tmt.base.Plan', 'tmt.base.Story']] = []
```

Comment:
`tmt.base.Core` would make the annotation simpler, unless there's a need for enumerating its child classes.

Reply:
There was a issue in mypy throwing an error `List item 0 has incompatible type "list[Test]"; expected "list[Core]"` when the `tmt.base.Core` annotation was used, that is why it was changed to the Union

Reply:
I'd give it a try, we should be able to address that issue in another way.

Reply:
@happz Any suggestions for the mentioned other way for how to resolve it? :)  
Mypy doesn't like it being in list:  
```
incompatible type "list[Plan]"; expected "list[Core]"  : note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
```

Reply:
> `Sequence` will not work, because it's read-only, maybe a `cast`?
> 
> ```python
> nodes: list[tmt.base.Core] = []
> for name in names:
>     if context.obj.tree.tests(names=[name]):
>         nodes.extend(cast(list[tmt.base.Core], context.obj.tree.tests(names=[name])))
> ```

no bueno
```
tmt/base.py:1302: error: Argument "nodes" to "jira_link" has incompatible type "list[Test]"; expected "list[Core]"  [arg-type]
tmt/base.py:1302: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tmt/base.py:1302: note: Consider using "Sequence" instead, which is covariant
```

Reply:
It's not that ugly in the end, no time to investigate more, we can simplify later if desired.